std/utf.d

```diff
@@ -2729,7 +2729,8 @@ private P toUTFzImpl(P, S)(S str) @safe pure
     {
         typeof(*P.init)[] retval = ['\0'];
 
-        return retval.ptr;
+        auto trustedPtr() @trusted { return retval.ptr; }
+        return trustedPtr();
```

Comment:
I'm not saying it's awful and should cause the PR not to be pulled. What I'm saying is that it seems unnecessarily open for abuse. I could see a refactor at some point somehow open it up to the rest of the function.

It's also more verbose. If we can do better, shouldn't we?